code/render/3dmath.cpp

```diff
@@ -225,20 +225,25 @@ int g3_project_vertex(vertex *p)
 
 	if ( p->flags & PF_PROJECTED )
 		return p->flags;
+	
+	w = 1.0f / p->world.xyz.z;
```

Comment:
Might this need a z == 0 safeguard?

Reply:
I guess? It didn't before obviously, and it *is* float math, for which we shouldn't expect 0 to naturally crop up. Should we just put in some arbitrarily big number in that case?

Reply:
IMO, all three components of xyw should probably be zero in that case

Reply:
Although, returning 0,0 would make our subsequent `atan2()` call *also* divide by 0...

Reply:
I mean, it's using atan2_safe, so _should_ be fine

Reply:
Oh. True.